poh/src/poh_recorder.rs

```diff
@@ -308,13 +308,15 @@ pub struct PohRecorder {
     ticks_from_record: u64,
     last_metric: Instant,
     record_sender: Sender<Record>,
+    pub leader_bank_status: Arc<LeaderBankStatus>,
     pub is_exited: Arc<AtomicBool>,
 }
 
 impl PohRecorder {
     fn clear_bank(&mut self) {
         if let Some(working_bank) = self.working_bank.take() {
             let bank = working_bank.bank;
+            self.leader_bank_status.set_completed(bank.slot());
```

Comment:
i'd rewrite this a bit to signify these 3 lines (esp, `.take()`-then-immediately-`.set_completed()` relationship) are strictly coupled semantically:

```rust
if let Some(WorkingBank { bank, start, .. }) = self.working_bank.take() {
  self.leader_bank_status.set_completed(bank.slot());
  ...
  datapoint_info!(
      "leader-slot-start-to-cleared-elapsed-ms",
      ("elapsed", start.elapsed().as_millis(), i64),
   ...  
}
```

Reply:
Makes sense - done: 5995d31d70345e80551af84e7117047d25af66f0